db/import_listener.go

```diff
@@ -156,7 +157,24 @@ func (il *importListener) ProcessFeedEvent(event sgbucket.FeedEvent) (shouldPers
 	return true
 }
 
+// calculateImportCompute calculates the import Process compute stat. NOTE: this current uses MiB as the MB unit to compute the stat
+// to use MB (Si) unit we must change the 'MiB' constant used here to 1000000
+func calculateImportCompute(bytes, functionTime float64) float64 {
+	timeMS := functionTime * float64(1000)
+	eventMb := bytes / base.MiB
+	stat := timeMS * (eventMb / 32 * base.MiB)
+	return stat
+}
+
 func (il *importListener) ImportFeedEvent(event sgbucket.FeedEvent) {
+	startTime := time.Now()
+	defer func() {
+		// we must grab the time in seconds here and convert to ms as the .Milliseconds() function returns integer millisecond count
+		functionTime := time.Since(startTime).Seconds()
+		bytes := float64(len(event.Value))
+		stat := calculateImportCompute(bytes, functionTime)
+		il.importStats.ImportProcessCompute.Add(stat)
+	}()
```

Comment:
you only want to actually calculate stats if this is going to run the import function. We have to unmarshal the document or the xattrs to see if it has metadata before we know if we need to run import.

so it has to hit `syncData == nil || !isSGWrite` block. I think this is why the import double counting occurs with on demand access. So probably set a variable for whether it is actually running an import and only report in the defer if import was run.